src/Tempest/Http/src/GenericRouter.php

```diff
@@ -45,7 +43,7 @@ public function dispatch(Request|PsrRequest $request): Response
             $request = map($request)->with(RequestToPsrRequestMapper::class);
         }
 
-        $matchedRoute = $this->matchRoute($request);
+        $matchedRoute = $this->routeMatcher->match($request);
```

Comment:
I extracted all matching logic to a new service "RouteMatcher". So  any logic for matching, static routes, dynamic routes and resolving parameters is moved to this new service.